### PR TITLE
Reformat with scalafmt

### PR DESCRIPTION
**Pull request by jvican**
The following commit follows the formatting convention established upstream in sbt/sbt.

### CODE REVIEW

**Comment by jvican** [comment]
I suggest to merge this after the open PRs to avoid unnecessary work to contributors.

**Comment by dwijnand** [comment]
.scalafmt.conf is missing

File: internal/compiler-bridge/src/main/scala/xsbt/Command.scala [hunk 1]
```diff
@@ -7,21 +7,31 @@
 
 package xsbt
 
-import scala.tools.nsc.{ CompilerCommand, Settings }
+import scala.tools.nsc.{CompilerCommand, Settings}
```
Comment (dwijnand) [inline internal/compiler-bridge/src/main/scala/xsbt/Command.scala]: This (and other examples) don't convince me we're really using the same configuration here..

**Review by dwijnand — CHANGES_REQUESTED** [review]

**Comment by jvican** [comment]
> .scalafmt.conf is missing
  
Oh shoot, forgot to add it when rebasing from another branch.

**Comment by jvican** [comment]
I've just updated the PR, thanks for the catch @dwijnand.

File: internal/compiler-bridge/src/main/scala/xsbt/DelegatingReporter.scala [hunk 2]
```diff
@@ -27,29 +36,33 @@ private object DelegatingReporter {
     override def toString =
       (sourcePath0, line0) match {
         case (Some(s), Some(l)) => s + ":" + l
-        case (Some(s), _)       => s + ":"
-        case _                  => ""
+        case (Some(s), _) => s + ":"
+        case _ => ""
```
Comment (dwijnand) [inline internal/compiler-bridge/src/main/scala/xsbt/DelegatingReporter.scala]: why are these not aligned?

NB when changing scalafmt config you can't just reformat again - you need to format from the original source code, then the result will correctly reflect the specified configuration.

**Review by dwijnand — CHANGES_REQUESTED** [review]

File: internal/compiler-bridge/src/main/scala/xsbt/DelegatingReporter.scala (hunk 2 above)
Comment (jvican) [reply internal/compiler-bridge/src/main/scala/xsbt/DelegatingReporter.scala]: The last commit aligns them :smile:

File: internal/zinc-apiinfo/src/main/scala/xsbt/api/APIUtil.scala [hunk 3]
```diff
@@ -13,7 +13,10 @@ object APIUtil {
   val modifiersToByte = (m: Modifiers) => {
     import m._
     def x(b: Boolean, bit: Int) = if (b) 1 << bit else 0
-    (x(isAbstract, 0) | x(isOverride, 1) | x(isFinal, 2) | x(isSealed, 3) | x(isImplicit, 4) | x(isLazy, 5) | x(isMacro, 6)).toByte
+    (x(isAbstract, 0) | x(isOverride, 1) | x(isFinal, 2) | x(isSealed, 3) | x(isImplicit, 4) | x(
+      isLazy,
+      5
+    ) | x(isMacro, 6)).toByte
```
Comment (dwijnand) [inline internal/zinc-apiinfo/src/main/scala/xsbt/api/APIUtil.scala]: lol

File: internal/zinc-apiinfo/src/main/scala/xsbt/api/Discovery.scala [hunk 4]
```diff
@@ -25,33 +27,44 @@ class Discovery(baseClasses: Set[String], annotations: Set[String]) {
...
+    d(definitions)
+  }
   def applications(definitions: Seq[Definition]): Seq[(Definition, Discovered)] =
     apply(Set.empty, Set.empty)(definitions)
 
   def findAnnotations(as: Seq[Annotation], pred: String => Boolean): Set[String] =
-    as.flatMap { a => simpleName(a.base).filter(pred) }.toSet
+    as.flatMap { a =>
+      simpleName(a.base).filter(pred)
+    }.toSet
```
Comment (dwijnand) [inline internal/zinc-apiinfo/src/main/scala/xsbt/api/Discovery.scala:57]: Is this a scalafmt bug? It's sub 100 columns (it's 62). /cc @olafurpg

File: internal/zinc-compile-core/src/main/scala/sbt/internal/inc/LoggerReporter.scala [hunk 5]
```diff
@@ -53,7 +53,14 @@ object LoggerReporter {
   lazy val problemStringFormats: ProblemStringFormats = new ProblemStringFormats {}
 }
 
-class LoggerReporter(maximumErrors: Int, logger: ManagedLogger, sourcePositionMapper: Position => Position = { p => p }) extends Reporter {
+class LoggerReporter(
+  maximumErrors: Int,
+  logger: ManagedLogger,
+  sourcePositionMapper: Position => Position = { p =>
+    p
+  }
```
Comment (dwijnand) [inline internal/zinc-compile-core/src/main/scala/sbt/internal/inc/LoggerReporter.scala]: This is weird too. 57 cols < 100 cols. /cc @olafurpg

**Review by dwijnand — CHANGES_REQUESTED** [review]
Added some random comments, mostly for Olaf.

But the thing missing here for me is CI enforcement of the format.

**Comment by jvican** [comment]
> But the thing missing here for me is CI enforcement of the format.
  
There is CI enforcement of the format via `scalafmtCheck`. But users will need to make sure all sources are compiled by running scalafmt manually or via their IDE.

File: internal/zinc-apiinfo/src/main/scala/xsbt/api/APIUtil.scala (hunk 3 above)
Comment (olafurpg) [reply internal/zinc-apiinfo/src/main/scala/xsbt/api/APIUtil.scala]: See http://scalameta.org/scalafmt/#Infixapplications line breaks around infix operators are your responsibility, so in this case scalafmt won't insert a line break before or after `|`

File: internal/zinc-apiinfo/src/main/scala/xsbt/api/Discovery.scala (hunk 4 above)
Comment (olafurpg) [reply internal/zinc-apiinfo/src/main/scala/xsbt/api/Discovery.scala:57]: Scalafmt is too eager at breaking down curly brace lambdas. I've known about this for a long time and iirc someone complained about it on gitter sometime but it has never been reported. I suspect it's not too hard to fix, so I invite you to open an issue ;)

However,  scalafmt will not try to move a curly brace lambda into a single line if there already is a newline inside the block.

File: internal/zinc-compile-core/src/main/scala/sbt/internal/inc/LoggerReporter.scala (hunk 5 above)
Comment (olafurpg) [reply internal/zinc-compile-core/src/main/scala/sbt/internal/inc/LoggerReporter.scala]: Same as above, scalafmt doesn't seem to like single line curly brace lambdas at all 😅

File: internal/zinc-apiinfo/src/main/scala/xsbt/api/APIUtil.scala (hunk 3 above)
Comment (dwijnand) [reply internal/zinc-apiinfo/src/main/scala/xsbt/api/APIUtil.scala]: Oh I know the problem. I'm sure over the coming time while I work on a codebase with scalafmt enforcement (:tada:) I'll get motivated enough to try and contribute an improvements in this area to scalafmt :)

File: internal/zinc-apiinfo/src/main/scala/xsbt/api/Discovery.scala (hunk 4 above)
Comment (dwijnand) [reply internal/zinc-apiinfo/src/main/scala/xsbt/api/Discovery.scala:57]: https://github.com/scalameta/scalafmt/issues/913

**Comment by dwijnand** [comment]
`scalafmtCheck` just runs git, so nothing is enforcing the format, right? Or am I missing something?

**Comment by jvican** [comment]
> scalafmtCheck just runs git, so nothing is enforcing the format, right? Or am I missing something?
  
Gimme a sec, doing changes to the whole design :wink:.

**Comment by jvican** [comment]
I've just updated this PR with the same changes as https://github.com/sbt/sbt/pull/3141.

**Comment by jvican** [comment]
This has been completely updated as per https://github.com/sbt/librarymanagement/pull/87. Please review.

File: build.sbt [hunk 6]
```diff
@@ -75,6 +83,7 @@ def altPublishSettings: Seq[Setting[_]] = Seq(
   })
 
 lazy val zincRoot: Project = (project in file(".")).
+  disablePlugins(com.typesafe.sbt.SbtScalariform).
```
Comment (eed3si9n) [inline build.sbt]: Why don't we just remove scalariform from the build?

**Review by eed3si9n — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 6 above)
Comment (jvican) [reply build.sbt]: Because it's not directly in the build, sbt house-rules depends on it.

Comment (eed3si9n) [reply build.sbt]: I see. That makes sense.

**Review by eed3si9n — APPROVED** [review]

**Comment by eed3si9n** [comment]
LGTM

**Comment by eed3si9n** [comment]
I've opened https://github.com/sbt/sbt-houserules/issues/15

**Review by dwijnand — CHANGES_REQUESTED** [review]
This is missing CI enforcement.

**Comment by jvican** [comment]
Well spotted @dwijnand. I had it in unstaged! Just pushed.

**Comment by eed3si9n** [comment]
```
[error] Expected configuration
[error] Expected ':' (if selecting a configuration)
[error] Expected key
[error] Expected '::'
[error] Expected end of input.
[error] zincScripted/test zinc/test zincPersist/test zincIvyIntegration/test zincTesting/test scalafmtCheck
[error]                  ^
```

**Review by dwijnand — APPROVED** [review]
Good to go.

**Pull request merged**